app/scripts/directives/loading.dir.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+angular.module('yourStlCourts').directive('loading', function($timeout){
+  return{
+    restrict: 'E',
+    scope:{
+    },
+    templateUrl: 'views/loading.html',
+    link: function($scope,element,attrs){
+      var fastEnough = true;
+      $(element).addClass("hidden");
+      $scope.$on('$stateChangeStart',function(){
+        fastEnough = false;
```

Comment:
Can you trigger multiple simultaneous state changes in Angular? Can state changes fail?